src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -648,6 +652,32 @@ OverlayImpl::getActivePeers()
     return ret;
 }
 
+void
+OverlayImpl::checkSanity (std::uint32_t index)
+{
+    std::lock_guard <decltype(mutex_)> lock (mutex_);
+
+    for (auto const& e : m_publicKeyMap)
+    {
+        auto const sp = e.second.lock();
+        if (sp)
+            sp->checkSanity (index);
+    }
+}
+
+void
+OverlayImpl::check ()
+{
+    std::lock_guard <decltype(mutex_)> lock (mutex_);
+
+    for (auto const& e : m_publicKeyMap)
+    {
+        auto const sp = e.second.lock();
```

Comment:
This is almost a textbook use of:
    if (auto const sp = e.second.lock())
        sp->check();